simde/x86/sse4.2.h

```diff
@@ -92,6 +92,283 @@ SIMDE_BEGIN_DECLS_
   #define _SIDD_UNIT_MASK SIMDE_SIDD_UNIT_MASK
 #endif
 
+SIMDE_FUNCTION_ATTRIBUTES
+int
+simde_mm_cmpestra_8_(simde__m128i a, int la, simde__m128i b, int lb, const int imm8)
+  SIMDE_REQUIRE_CONSTANT_RANGE(imm8, 0, 255) {
+  const int cmp_op = imm8 & 0x0c;
+  const int polarity = imm8 & 0x30;
+  simde__m128i_private
+    bool_res_ = simde__m128i_to_private(simde_mm_setzero_si128()),
+    a_ = simde__m128i_to_private(a),
+    b_ = simde__m128i_to_private(b);
+  const int upper_bound = (128 / 8) - 1;
+  int a_invalid = 0;
+  int b_invalid = 0;
+  for(int i = 0 ; i <= upper_bound ; i++) {
+    for(int j = 0; j <= upper_bound ; j++){
+      int bitvalue = ((a_.i8[i] == b_.i8[j]) ? 1 : 0);
+      if(i == la)
+        a_invalid = 1;
+      if(j == lb)
+        b_invalid = 1;
+      switch(cmp_op){
+        case SIMDE_SIDD_CMP_EQUAL_ANY:
+        case SIMDE_SIDD_CMP_RANGES:
+          bitvalue = 0;
+          break;
+        case SIMDE_SIDD_CMP_EQUAL_EACH:
+          if(a_invalid && b_invalid)
+            bitvalue = 1;
+          else
+            bitvalue = 0;
+          break;
+        case SIMDE_SIDD_CMP_EQUAL_ORDERED:
+          if(a_invalid && !b_invalid)
+            bitvalue = 1;
+          else if(a_invalid && b_invalid)
+            bitvalue = 1;
+          else
+            bitvalue = 0;
+          break;
+      }
+      if(bitvalue)
+        bool_res_.i16[i] |= (1UL << j);
+      else
+        bool_res_.i16[i] &= ~(1UL << j);
+    }
+  }
+  int32_t int_res_1 = 0;
+  int32_t int_res_2 = 0;
+  switch(cmp_op) {
+    case SIMDE_SIDD_CMP_EQUAL_ANY:
+      for(int i = 0 ; i <= upper_bound ; i++){
+        SIMDE_VECTORIZE_REDUCTION(|:int_res_1)
+        for(int j = 0 ; j <= upper_bound ; j++){
+          int_res_1 |= (((bool_res_.i8[i] >> j) & 1) << i);
+        }
+      }
+      break;
+    case SIMDE_SIDD_CMP_RANGES:
+      for(int i = 0 ; i <= upper_bound ; i++){
+        SIMDE_VECTORIZE_REDUCTION(|:int_res_1)
+        for(int j = 0 ; j <= upper_bound ; j++){
+          int_res_1 |= ((((bool_res_.i8[i] >> j) & 1) & ((bool_res_.i8[i] >> (j + 1)) & 1)) << i);
+          j += 2;
+        }
+      }
+      break;
+    case SIMDE_SIDD_CMP_EQUAL_EACH:
+      for(int i = 0 ; i <= upper_bound ; i++){
+        //SIMDE_VECTORIZE_REDUCTION(|:int_res_1)
```

Comment:
what should I do here, if I have to use 2 operators inside this loop?

Reply:
You can't have two reductions on the same variable.

That said, I don't see why you need this; `int_res_1` is initialized to 0 (all bits unset), so you shouldn't need to unset the bits here.  You only need to handle the case where the bits should be set but aren't.